skunkworks_crow/src/main/java/org/odk/share/services/ReceiverService.java

```diff
@@ -52,15 +52,29 @@ private void addDownloadJobSubscription() {
                             break;
                     }
                 }).subscribe();
+    }
 
+    public void startBtDownloading(String macAddress) {
+        PersistableBundleCompat extras = new PersistableBundleCompat();
+        extras.putString("mac", macAddress);
+        extras.putInt("MODE_OF_TRANSFER", Share.TransferMethod.BLUETOOTH);
+        JobRequest request = new JobRequest.Builder(DownloadJob.TAG)
```

Comment:
Why not pass the `extras` to `startJob()` method? Also, like I mentioned earlier that would resolve the problem that this job doesn't trigger a rx bus event

Reply:
improved.